MM-10703 Adding blank request context to plugin hooks for future use.

crspeller opened the pull request:
A temporary solution to avoid breaking changes when we add this for real.

jwilander commented:
Should we add the context to the `ServeHTTP` hook as well?

lieut-data commented on plugin/request_context.go:
```diff
@@ -0,0 +1,7 @@
+
+package plugin
+
+type RequestContext struct {
```
nit: What's the thinking re: `RequestContext` vs just `Context`?

crspeller commented:
@lieut-data @jwilander 
- Renamed RequestContext to Context
- added context to ServeHTTP and ExecuteCommand
- fixed the import cycle.

Ready for re-review.
